packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -308,10 +308,25 @@ export class ProjectLifecycleManager {
   /**
    * Sets the initial `activeBrowser` depending on these criteria, in order of preference:
    *  1. The value of `--browser` passed via CLI.
-   *  2. The last browser selected in `open` mode (by name and channel) for this project.
-   *  3. The first browser found.
+   *  2. The value of `defaultBrowser` in `cypress.config`.
+   *  3. The last browser selected in `open` mode (by name and channel) for this project.
+   *  4. The first browser found.
    */
   async setInitialActiveBrowser () {
+    const configBrowser = this.loadedFullConfig?.defaultBrowser
+
+    // if we have a defaultBrowser from the config and a CLI browser wasn't passed and,
+    // the active browser isn't set (we are in run mode or when the config changes in open mode),
+    // update the cliBrowser to the defaultBrowser from the config
+    // since we want the defaultBrowser to behave as if it was passed via CLI
+    if (configBrowser && !this.ctx.modeOptions.isBrowserGivenByCli) {
+      if (!this.ctx.coreData.activeBrowser) {
+        this.ctx.actions.browser.updateCliBrowser(configBrowser)
+      } else {
+        this.ctx.actions.browser.updateCliBrowser(this.ctx.coreData.activeBrowser.name)
```

Comment:
What's the issue this change is solving?

Reply:
before this change...
with `defaultBrowser` set (and --browser not given) in `open` mode, changing the browser in the GUI after the fact and the saving the config file - the name of the changed browser is wrong